File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Arbeidsland.java
package no.nav.melosys.eessi.controller.dto;

import lombok.Data;

import java.util.List;

@Data
public class Arbeidsland {

    private String land;
    private List<Arbeidssted> arbeidssted;
    private String harfastarbeidssted;

    public static Arbeidsland av(no.nav.melosys.eessi.models.sed.nav.Arbeidsland arbeidslandFraRina) {
        Arbeidsland arbeidsland = new Arbeidsland();

        arbeidsland.land = arbeidslandFraRina.getLand();
        arbeidsland.arbeidssted = arbeidslandFraRina.getArbeidssted().stream().map(Arbeidssted::av).toList();
        arbeidsland.harfastarbeidssted = arbeidsland.getHarfastarbeidssted();

        return arbeidsland;
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/controller/dto/Arbeidsland.kt
package no.nav.melosys.eessi.controller.dto

import no.nav.melosys.eessi.models.sed.nav.Arbeidsland as RinaArbeidsland

data class Arbeidsland(
    var land: String? = null,
    var arbeidssted: List<Arbeidssted> = emptyList()
) {
    companion object {
        fun av(arbeidslandFraRina: RinaArbeidsland): Arbeidsland = Arbeidsland(
            land = arbeidslandFraRina.land,
            arbeidssted = arbeidslandFraRina.arbeidssted.map(Arbeidssted::av)
        )
    }
}
